### ISSUE TITLE
[PW-6588] Consecutive order history comments

### ISSUE THREAD

**Issue by jonathanribas** [Adyen/adyen-magento2#1473]
**Is your feature request related to a problem? Please describe.**
Adobe Commerce 2.4.3-p1
Adyen module: 8.2.0

Sinc Adyen `8.0.0` we have noticed some errors on our order workflow: Adyen  / Signifyd / Avatax.

Before `8.0.0`, on `7.2.0`, you were forcing order status when adding comment to orders: https://github.com/Adyen/adyen-magento2/pull/1104/files. It seems since you have refactored this code, it has broken something. Maybe you were "fixing" an Adobe Commerce native issue.

When order full amount is finalized, webhook is adding 2 consecutive comments to order history (1 source code line separate the 2 comments).

First comment is supposed to not change order status but it's actually reverting it to `payment_review`, then second one update it to another status if conditions are met.

CAPTURE webhook notification w/amount xxxx was processed:
https://github.com/Adyen/adyen-magento2/blob/599f4d3b88168a4f406d0fc5dd757c058a3dc620/Helper/Webhook.php#L1651

Adyen Payment Successfully completed: 
https://github.com/Adyen/adyen-magento2/blob/599f4d3b88168a4f406d0fc5dd757c058a3dc620/Helper/Webhook.php#L1652

`sales_order_status_history` extract that shows order status update to an old one `payment_review`:

<img width="1155" alt="Screenshot 2022-04-23 at 07 03 03" src="https://user-images.githubusercontent.com/3003782/164878529-f20d23ea-ac25-490a-8270-6cdfb69ed56d.png">

**Describe the solution you'd like**
It seems `addStatusHistoryComment` from `vendor/magento/module-sales/Model/Order.php` is deprecated since `101.0.5`.

Maybe you should consider using `setState` sending `comment` in method parameters from `vendor/magento/module-sales/Model/Service/OrderService.php`. But in the end this method uses `addStatusHistoryComment` too. Or force current order status as it was done on https://github.com/Adyen/adyen-magento2/pull/1104/files

In conclusion, comments shouldn't set order to another status to avoid any workflow issues on merchants.

**Comment by tnaber** [comment]
Hi @jonathanribas,

Thank you for opening this issue, we will be looking into this. We have also noticed some unwanted behaviour where the order is closed instead of completed, this might be correlated.

Cheers,
Titus

**Comment by jonathanribas** [comment]
Hi @tnaber, yes, it's what I've commented in my last comment https://github.com/Adyen/adyen-magento2/issues/1473#issuecomment-1117576217

**Comment by tnaber** [comment]
Hi @jonathanribas, I was also thinking about the discussion here: https://github.com/Adyen/adyen-magento2/pull/1390#issuecomment-1095170677 if you're interested ;)

**Comment by Morerice** [comment]
Hi @jonathanribas,

Since we made quite a big refactoring change regarding how webhooks are handled in v8.3.3, would it be possible to check if this issue is still present in this version (or more recent ones)?

Thanks,
Jean
Adyen

**Comment by jonathanribas** [comment]
Hi @Morerice, we will go live this week on `8.3.5`, I will let you know if we still see those consecutive order history comments.

**Comment by Morerice** [comment]
Hi @jonathanribas,

Thanks for the update. For now I will close the issue. However if this problem persists, feel free to re-open it and we can continue our discussion.

Thanks,
Jean
Adyen
